### DISCUSSION TITLE
Hardware timestamping analog input events

### DISCUSSION THREAD

**Discussion by ZegCricket** [ni/nidaqmx-python#596] [Q&A] [answered]
Hello,

I'm using a NI USB-6210 to acquire some analog data at a sampling frequency of 250 kHz and I wanted to know if it is possible to timestamp the events in hardware. I know it is possible to timestamp the events (or the beginning and the end of the task.read) in software, but I don't know how precise the time measurements can be if done in that way.

Thank you in advance for your help!

**Comment by zhindes** [comment, marked as answer]
The hardware does not have any timestamping capabilities, no. However, if you do timestamp the start of an acquisition in software (e.g. when you call `task.start()`), assuming you have no hardware start trigger, then you can reasonably extrapolate the time of the data from that. This what the NI-DAQmx driver does for timestamping data for use in LabVIEW waveforms. Purely software timestamping and extrapolation.

**Reply by ZegCricket** [reply]
Thank you so much! :)

**Comment by pbellino** [comment]
You have two counters available. Route your 80 MHz clock signal to one of the counters.Use the other counter as the gate for that signal.
When the gate is high, the counter will count down a specified number of 80 MHz clock pulses.Your result will be in “clock pulses.”
To convert to time, multiply the pulse count by the clock period (12.5 ns for 80 MHz).
